File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    memory = {}
    
    for k in keymap: # keymap을 완전 탐색하면서 최소 인덱스가 무엇인지 확인하기, memory 업데이트
        for i in range(len(k)):
            memory[k[i]] = (i + 1) if k[i] not in memory else min(memory[k[i]], i+1)
    
    for t in targets:
        cnt = 0
        for i in range(len(t)):
            if t[i] in memory:
                cnt += memory[t[i]]
            else:
                cnt = -1
                break
        answer.append(cnt)

    return answer

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

vector<int> solution(vector<string> keymap, vector<string> targets) {
    vector<int> answer;
    unordered_map<char, int> um;
    
    for(string k: keymap){
        for(int i=0; i < k.length(); i++){
            if(um.find(k[i]) == um.end()){
                um[k[i]] = i+1;
            }
            else{
                um[k[i]] = min(i+1, um[k[i]]);
            }
        }
    }
    
    for(string t: targets){
        int cnt = 0;
        for(char c: t){
            if(um.find(c) != um.end()){
                cnt += um[c];
            } else{
                cnt = -1;
                break;
            }
        }
        answer.emplace_back(cnt);
    }
    return answer;
}